app/helpers/physical_server_helper/textual_summary.rb

```diff
@@ -91,15 +91,24 @@ def textual_power_state
   end
 
   def textual_mac
-    {:label =>  _("Mac Address"), :value => @record.hardware.guest_devices.collect { |device| device[:address] }.join(", ") }
+    # Currently, we only want to display the MAC addresses of devices that
+    # are associated with an entry in the networks table. This ensures that
+    # we only display management addresses.
+    {:label =>  _("MAC Address"), :value => @record.hardware.guest_devices.reject { |device| device.network.nil? }.collect { |device| device[:address] }.join(", ") }
   end
 
   def textual_ipv4
-    {:label =>  _("IPV4 Address"), :value => @record.hardware.guest_devices.collect { |device| device.network.ipaddress }.join(", ") }
+    # It is possible for guest devices not to have network data (or a network
+    # hash). As a result, we need to exclude guest devices that don't have
+    # network data to prevent a nil class error from occurring.
+    {:label =>  _("IPv4 Address"), :value => @record.hardware.guest_devices.reject { |device| device.network.nil? }.collect { |device| device.network.ipaddress }.join(", ") }
```

Comment:
Good point. 
I would have suggested `["10.243.6.17", "].compact.join(', ')`, but I think I like your approach better. 

So, let's leave things the way they are. Thanks.